packages/react-menu/src/components/MenuList/index.ts

```diff
@@ -1,5 +1,14 @@
 export * from './MenuList';
-export * from './MenuList.types';
+// Explicit exports to ommit MenuListCommon
+export type {
+  MenuCheckedValueChangeEvent,
+  MenuCheckedValueChangeData,
+  MenuListSlots,
+  MenuListProps,
+  MenuListState,
+  MenuListContextValues,
+  UninitializedMenuListState,
+} from './MenuList.types';
```

Comment:
Could we remove the `export` keyword from the Commons types instead of having index.ts list all of the non-Commons exports? Otherwise, we'll have to come back and fix the exports once we actually remove the Commons types.

I did quick search, and it looks like importing the Commons types is uncommon (so to speak). At least, the only place I could find other than the Badge.stories.tsx you're fixing, is here for MenuListCommons. You could make the index.ts change only here, and add a TODO comment on MenuListCommons, etc., that the index.ts file needs to be fixed up once the Commons type is removed.